miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -21,25 +21,24 @@ internal class MiniAppDownloader(
 
     @Suppress("SwallowedException", "LongMethod")
     suspend fun getMiniApp(appId: String, versionId: String): String {
-        val versionPath = storage.getMiniAppVersionPath(appId, versionId)
         try {
+            val miniAppInfo = apiClient.fetchInfo(appId)
+            val versionPath = storage.getMiniAppVersionPath(miniAppInfo.id, miniAppInfo.version.versionId)
             return when {
-                !isLatestVersion(appId, versionId) -> throw sdkExceptionForInvalidVersion()
-                miniAppStatus.isVersionDownloaded(appId, versionId, versionPath) -> versionPath
-                else -> startDownload(appId, versionId)
+                miniAppStatus.isVersionDownloaded(
+                    miniAppInfo.id, miniAppInfo.version.versionId, versionPath) -> versionPath
+                else -> startDownload(miniAppInfo.id, miniAppInfo.version.versionId)
             }
         } catch (netError: MiniAppNetException) {
             // load local if possible when offline
+            val versionPath = storage.getMiniAppVersionPath(appId, versionId)
```

Comment:
This means that the host app needs to know the last downloaded `versionId` if they want to launch a Mini App while offline. So they would need to cache the `MiniAppInfo` object?

I think there are two possible solutions to this:
1. Add another function: `MiniApp#create(miniAppId: String, miniAppMessageBridge: MiniAppMessageBridge)`. This function would do basically the same thing as the current `create` function, except that when the network is offline it would find the currently downloaded `versionId` for the Mini App and then load that. (We could also possibly deprecate the old `create` function - I think the original purpose of passing `MiniAppInfo` is that we thought there could be multiple versions of a Mini App published at the same time, but this isn't possible).
2. or Change `MiniApp#fetchInfo` so that it will return cached a cached `MiniAppInfo` object while the network is offline. But this would mean we need to keep this cache on the SDK side, so probably this option isn't ideal.

What do you think @minh-rakuten @ts-nitish-a-jain?

Reply:
I had updated with a solution no.1 👍 and also declare MiniApp#create(MiniAppInfo, MiniAppMessageBridge) as deprecated.